pkg/controller/infra/infra_controller.go

```diff
@@ -118,99 +138,369 @@ type InfraController struct {
+			clusterProvisioned.Status = kapi.ConditionTrue
+			clusterProvisioned.LastTransitionTime = now
+			clusterProvisioned.LastProbeTime = now
+			clusterProvisioned.Reason = "JobCompleted"
+			clusterProvisioning.Message = fmt.Sprintf("Job %s/%s completed at %v", job.Namespace, job.Name, jobCompleted.LastTransitionTime)
+			cluster.Status.Provisioned = true
+		}
+		if clusterProvisioned == nil {
+			cluster.Status.Conditions = append(cluster.Status.Conditions, clusteroperator.ClusterCondition{
+				Type:               clusteroperator.ClusterHardwareProvisioned,
+				Status:             kapi.ConditionTrue,
+				LastProbeTime:      now,
+				LastTransitionTime: now,
+				Reason:             "JobCompleted",
+				Message:            fmt.Sprintf("Job %s/%s completed at %v", job.Namespace, job.Name, jobCompleted.LastTransitionTime),
+			})
+			cluster.Status.Provisioned = true
+		}
+		cluster.Status.ProvisioningJob = nil
+	}
+	jobFailed := jobCondition(job, v1batch.JobFailed)
+	if jobFailed != nil {
+		clusterProvisioning := clusterCondition(cluster, clusteroperator.ClusterHardwareProvisioning)
+		if clusterProvisioning != nil &&
+			clusterProvisioning.Status == kapi.ConditionTrue {
+			clusterProvisioning.Status = kapi.ConditionFalse
+			clusterProvisioning.LastTransitionTime = now
+			clusterProvisioning.LastProbeTime = now
+			clusterProvisioning.Reason = "JobFailed"
+			clusterProvisioning.Message = fmt.Sprintf("Job %s/%s failed at %v, reason: %s", job.Namespace, job.Name, jobFailed.LastTransitionTime, jobFailed.Reason)
+		}
+		cluster.Status.ProvisioningJob = nil
+	}
+
+	if jobCompleted == nil && jobFailed == nil && job.Status.Active > 0 {
+		clusterProvisioning := clusterCondition(cluster, clusteroperator.ClusterHardwareProvisioning)
+		if clusterProvisioning != nil {
+			if clusterProvisioning.Status == kapi.ConditionFalse {
+				clusterProvisioning.Status = kapi.ConditionTrue
+				clusterProvisioning.LastTransitionTime = now
+			}
+			clusterProvisioning.LastProbeTime = now
+			clusterProvisioning.Reason = "JobRunning"
+			clusterProvisioning.Message = fmt.Sprintf("Job %s/%s is running since %v. Pod completions: %d, failures: %d", job.Namespace, job.Name, job.Status.StartTime, job.Status.Succeeded, job.Status.Failed)
+		} else {
+			cluster.Status.Conditions = append(cluster.Status.Conditions, clusteroperator.ClusterCondition{
+				Type:               clusteroperator.ClusterHardwareProvisioning,
+				Status:             kapi.ConditionTrue,
+				LastProbeTime:      now,
+				LastTransitionTime: now,
+				Reason:             "JobRunning",
+				Message:            fmt.Sprintf("Job %s/%s is running since %v. Pod completions: %d, failures: %d", job.Namespace, job.Name, job.Status.StartTime, job.Status.Succeeded, job.Status.Failed),
+			})
+		}
+	}
+
+	return c.updateClusterStatus(cluster)
+}
+
+func (c *InfraController) updateClusterStatus(cluster *clusteroperator.Cluster) error {
+	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
```

Comment:
I don't think that this is the right thing to do here. A conflict occurs because the controller has a cluster object that is out of date. The request will perpetually fail since the cluster object that the controller has will remain out of date. The controller is just wasting time attempting to retry the request.

Reply:
duh, yes, I do need to retry on conflict, but will need to do a get on the cluster resource

Reply:
Can we just fail the sync and pick up the correct cluster object on the next retry? That seems much easier than trying to recreate the changes to the cluster status with the update version. Of course, in that case we would need to handle looking for the job and configmap to make sure that we don't accidentally create multiples. So that may make it more difficult.

Reply:
Even if we did get the latest and retry inside a single sync call, we still have the potential to create multiple jobs, so doing it that way doesn't even buy us that simplification.